addons/sale_stock/models/sale_order.py

```diff
@@ -87,14 +87,10 @@ def _compute_delivery_status(self):
     @api.depends('picking_policy')
     def _compute_expected_date(self):
         super(SaleOrder, self)._compute_expected_date()
-        for order in self:
-            dates_list = []
-            for line in order.order_line.filtered(lambda x: x.state != 'cancel' and not x._is_delivery() and not x.display_type):
-                dt = line._expected_date()
-                dates_list.append(dt)
-            if dates_list:
-                expected_date = min(dates_list) if order.picking_policy == 'direct' else max(dates_list)
-                order.expected_date = fields.Datetime.to_string(expected_date)
+
+    def _select_expected_date(self, expected_dates):
+        res = super()._select_expected_date()
+        return res if self.picking_policy == 'direct' else max(expected_dates)
```

Comment:
```suggestion
        if self.picking_policy == 'direct:
            return super()._select_expected_date(expected_dates)
        return max(expected_dates)
```

You forgot to pass the arg to the super call.
Also, let's avoid computing the `min` if we know we will return the `max`.